---
title: "CDK4/6 and CH"
author: "Irenaeus Chan"
date: "`r format(Sys.time(), '%a %b %d, %Y %X')`"
output:
  html_document:
    code_folding: hide
    toc: true
    toc_depth: 2
    number_sections: true
    theme: cosmo
    df_print: kable
    toc_float:
      collapsed: false
      smooth_scroll: false
  github_document:
    toc: true
    toc_depth: 2
    fig_width: 10
    fig_height: 10
---

```{r setup, include=TRUE, message = FALSE, warning = FALSE}
knitr::opts_chunk$set(dev = "svg")

library(tidyverse)
library(data.table)
library(table1)
library(ggpubr)
library(ComplexHeatmap)
library(ggsci)
library(ggsignif)
library(cowplot)
library(ggplot2)
library(geepack)
library(sjPlot)
library(readxl)
library(maftools)
library(purrr)
library(RColorBrewer)

library(conflicted)
conflicts_prefer(dplyr::select)
conflicts_prefer(dplyr::filter)

source("utils/custom_lollipop.r")

data_file_path <- "data/"
figures_save_path <- "figures/"
```

```{r load_data, include=TRUE, message = FALSE, warning = FALSE}
# CH Calls
sclc_df <- fread(paste0(data_file_path, "sclc_df.minimum.csv"))                       # Small Cell Lung Cancer
tnbc_df <- fread(paste0(data_file_path, "tnbc_df.minimum.csv"))                       # Triple Negative Breast Cancer
crc_df <- fread(paste0(data_file_path, "crc_df.minimum.csv"))                         # Colorectal Cancer
untreated_df <- fread(paste0(data_file_path, "untreated_df.minimum.csv"))                # Untreated

g1_df <- rbind(
    sclc_df %>% mutate(Cohort = "SCLC") %>% mutate(DeID = as.character(DeID)), 
    tnbc_df %>% mutate(Cohort = "TNBC") %>% mutate(DeID = as.character(DeID), SampleDeID = as.character(SampleDeID)), 
    crc_df %>% mutate(Cohort = "CRC"),
    untreated_df %>% mutate(Cohort = "Untreated") %>% mutate(SampleDeID = as.character(SampleDeID))
)

sclc_schematic <- paste0(data_file_path, "G1_SCLC.png")
tnbc_schematic <- paste0(data_file_path, "G1_TNBC.png")
crc_schematic <- paste0(data_file_path, "G1_CRC.png")

# Timepoints Dataframe
sclc_tp <- fread(paste0(data_file_path, "sclc_timepoints.under17000.minimum.csv"))                  # Small Cell Lung Cancer
tnbc_tp <- fread(paste0(data_file_path, "tnbc_timepoints.under17000.minimum.csv"))                  # Triple Negative Breast Cancer
crc_tp <- fread(paste0(data_file_path, "crc_timepoints.under17000.minimum.csv"))                    # Colorectal Cancer
untreated_tp <- fread(paste0(data_file_path, "untreated_timepoints.minimum.csv"))                    # Untreated

g1_tp <- rbind(
    sclc_tp %>% mutate(Cohort = if_else(Trilaciclib == "Untreated", "Untreated", "SCLC")), 
    tnbc_tp %>% mutate(Cohort = "TNBC"), 
    crc_tp %>% mutate(Cohort = "CRC"),
    untreated_tp %>% mutate(Cohort = "Untreated")
)
# Demographics Dataframe
g1_demo <- fread(paste0(data_file_path, "g1_demo.csv"))

# Blood Counts Lab Results
g1_lab <- fread(paste0(data_file_path, "g1_labresults.csv"))

# Bioinformatics Figure 4 Validation Data
bio_fig4_data <- fread(paste0(data_file_path, "cdk46_ch_pilot_harvard.filtered.csv"))
```

# Check that all Timepoints Variants are IN the Dataframe and vice versa
```{r check_timepoints, include=TRUE, message = FALSE, warning = FALSE}
# All of the timepoint variants exist within the CH calls
setdiff(
    g1_tp %>% select(DeID, key, Cohort),
    g1_df %>% filter(key != "") %>% select(DeID, key, Cohort) %>% distinct()
)

# 141 CH Call variants are not analyzed because they may be noise due to low sequencing depths (17k)
setdiff(
    g1_df %>% filter(key != "") %>% select(DeID, key, Cohort) %>% distinct(),
    g1_tp %>% select(DeID, key, Cohort) %>% distinct()
) %>% select(Cohort) %>% table()
```

# Table 1
```{r table1}
label(g1_demo$Age) <- "Age at Treatment Start"
label(g1_demo$Sex) <- "Gender"
label(g1_demo$Race) <- "Race"
label(g1_demo$SmokingStatus) <- "Smoking Status"

for_table <- g1_demo %>% 
    filter(ifelse(Cohort == "TNBC", whichdraw == "PreTx" | whichdraw == "MidTx", whichdraw == "PreTx" | whichdraw == "PostTx")) %>%
    filter(DeID %in% c(g1_demo %>% 
                        filter(ifelse(Cohort == "TNBC", whichdraw == "PreTx" | whichdraw == "MidTx", whichdraw == "PreTx" | whichdraw == "PostTx")) %>%
                        group_by(Cohort, DeID) %>%
                        summarise(n = n()) %>%
                        filter(n > 1) %>%
                        pull(DeID))
    )
```

# Plotting Variables
```{r plot_variables, include=TRUE, message = FALSE, warning = FALSE}
panel_theme_basic <- theme_bw() + theme(
  panel.border = element_blank(),
  legend.title = element_blank(),
  legend.key.size = unit(5, "mm"),
  legend.text = element_text(size = 22),
  legend.position = "top",
  legend.direction = "horizontal",
  plot.subtitle = element_text(hjust = 0.5, size = 24),
  plot.title = element_text(face = "bold", hjust = 0, vjust = -2, size = 24),
  panel.grid.major = element_blank(),
  strip.background = element_blank(),
  strip.text = element_text(size = 22),
  axis.text.y = element_text(size = 22),
  axis.text.x = element_text(size = 22),
  axis.title = element_text(size = 24),
  axis.line = element_line(colour = "black"),
  plot.margin = margin(1, 1, 1, 1, "cm")
)

default_sig <- function(y_position, xmin, xmax, model_pvalue, model_stars, tip_length = 0.03, size = 1, textsize = 10, color = "black", show_values = TRUE, hide_ns = TRUE) {
  if (show_values){
    list(y_position = y_position, xmin = c(xmin), xmax = c(xmax), annotation = paste0("p = ", format(model_pvalue, digits = 5), model_stars), tip_length = tip_length, size = size, textsize = textsize, color = color)
  } else {
    if (model_stars == "") { model_stars <- "ns"}
    if (hide_ns & model_stars == "ns") { 
      list(y_position = y_position, xmin = c(xmin), xmax = c(xmax), annotation = "", tip_length = 0, size = 0, textsize = 0, color = "black")
    } else {
      list(y_position = y_position, xmin = c(xmin), xmax = c(xmax), annotation = model_stars, tip_length = tip_length, size = size, textsize = textsize, color = color)
    }
  }
}

default_sig_data <- function(data, y_position, xmin, xmax, model_pvalue, model_stars, tip_length = 0.03, size = 1, textsize = 10, color = "black", show_values = TRUE, hide_ns = TRUE) {
  if (show_values) {
    list(data = data, y_position = y_position, xmin = c(xmin), xmax = c(xmax), annotation = paste0("p = ", format(model_pvalue, digits = 5), model_stars), tip_length = tip_length, size = size, textsize = textsize, color = color)
  } else {
    if (model_stars == "") { model_stars <- "ns"}
    if (hide_ns & model_stars == "ns") { 
      list(data = data, y_position = y_position, xmin = c(xmin), xmax = c(xmax), annotation = "", tip_length = 0, size = 0, textsize = 0, color = "black")
    } else {
      list(data = data, y_position = y_position, xmin = c(xmin), xmax = c(xmax), annotation = model_stars, tip_length = tip_length, size = size, textsize = textsize, color = color)
    }
  }
}

add_significance_annotation <- function(p, label, pos, start, end, size_text = 10, size_line = 1) {
    p +
      annotate("text", x = (pos + 0.1), y = (start + end)/2, label = label, size = size_text) +
      annotate("segment", x = pos, xend = pos, y = start, yend = end, size = size_line) +
      annotate("segment", x = pos - 0.05, xend = pos + 0.008, y = start, yend = start, size = size_line) +
      annotate("segment", x = pos - 0.05, xend = pos + 0.008, y = end, yend = end, size = size_line)
}

signif.num <- function(x, ns = FALSE) {
  if (ns) {
    symbols = c("***", "**", "*", "ns")
  } else {
    symbols = c("***", "**", "*", "")
  }
  
  symnum(unlist(x), corr = FALSE, na = FALSE, legend = T,
         cutpoints = c(0, 0.001, 0.01, 0.05, 1), 
         symbols = symbols)
}
```

# GLM Function for Growth Rate
```{r glm_growth_rate, include=TRUE, message = FALSE, warning = FALSE}
model_growth_rate_class_fun <- function(df_for_glm) {
  purrr::cross_df(list(
    Gene_Class = c("DTA", "DDR"),
    comparison = c("all", "no_untreated")
  )) %>%
    pmap_dfr(function(Gene_Class, comparison) {
        print(paste0("Fitting for ", Gene_Class, " with comparison ", comparison))
      df <- df_for_glm %>%
        filter(Gene_Class == !!Gene_Class)
      if (comparison == "no_untreated") {
        df <- df %>%
          filter(Trilaciclib != "Untreated") %>%
          mutate(Trilaciclib = factor(Trilaciclib, levels = c("Trilaciclib", "Placebo")))
      }
      fit <- glm(formula = growth_rate ~ Trilaciclib + Age, data = df)
      sjPlot::get_model_data(fit, type = "est") %>%
        mutate(
          term = case_when(
            term == "TrilaciclibTrilaciclib" ~ "UntreatedVSTrilaciclib",
            term == "TrilaciclibPlacebo" ~ ifelse(comparison == "all", "UntreatedVSPlacebo", "TrilaciclibVSPlacebo"),
            TRUE ~ term
          ),
          Gene_Class = Gene_Class
        )
    }) %>%
    filter(term %in% c("UntreatedVSTrilaciclib", "UntreatedVSPlacebo", "TrilaciclibVSPlacebo"))
}
```

# Default Variables
```{r default_variables, include=TRUE, message = FALSE, warning = FALSE}
gene_list = c("DNMT3A", "TET2", "ASXL1", "TP53", "PPM1D", "CHEK2", "JAK2", "SRSF2", "SF3B1")
DDR_genes = c("TP53", "PPM1D", "CHEK2")
treatment_colors = c('Untreated' = '#BC3C29FF', 'Placebo' = '#E18727FF', 'Trilaciclib' = '#20854EFF')
```

# Basic Processing - Merging ALL Information Together
```{r processing, include=TRUE, message = FALSE, warning = FALSE}
# Adding Gene and Variant Information to Timepoints
g1_tp <- g1_tp %>%
    left_join(
        g1_df %>%
            dplyr::select(
                key, Gene,
                VariantClass, gene_aachange, oncoKB, oncoKB_reviewed,
                n.loci.vep, source.totals.loci, n.loci.truncating.vep, source.totals.loci.truncating, n.HGVSp, source.totals.p, n.HGVSc, source.totals.c, CosmicCount, heme_cosmic_count, myeloid_cosmic_count
            ) %>%
            rowwise() %>%
            mutate(non_na_count = sum(across(everything(), ~ !is.na(.)))) %>%
            ungroup() %>%
            group_by(key, Gene) %>%
            slice_max(order_by = non_na_count, with_ties = FALSE) %>%
            ungroup() %>%
            dplyr::select(-non_na_count),
        by = c("key", "Gene")
    )

# Adding Demographics Information to Timepoints ie. Age, Sex, Race, etc for Statistics
g1_tp <- g1_tp %>%
    left_join(
        g1_demo %>% 
            filter(whichdraw == "PreTx") %>%
            dplyr::select(DeID, Age, Sex, Race, Ethnicity, SmokingStatus)
    )

# Calculating change in blood counts
g1_lab_changes <- g1_lab %>% 
            pivot_wider(
                id_cols = c(Cohort, subject),
                names_from = c(Testcode, whichdraw),
                values_from = Labvalue
            ) %>%
            mutate(
                perc_BASO_of_WBC_PreTx = BASO_PreTx / WBC_PreTx,
                perc_EOS_of_WBC_PreTx = EOS_PreTx / WBC_PreTx,
                perc_LYM_of_WBC_PreTx = LYM_PreTx / WBC_PreTx,
                perc_MONO_of_WBC_PreTx = MONO_PreTx / WBC_PreTx,
                perc_NEUT_of_WBC_PreTx = NEUT_PreTx / WBC_PreTx,

                perc_BASO_of_WBC_PostTx = BASO_PostTx / WBC_PostTx,
                perc_EOS_of_WBC_PostTx = EOS_PostTx / WBC_PostTx,
                perc_LYM_of_WBC_PostTx = LYM_PostTx / WBC_PostTx,
                perc_MONO_of_WBC_PostTx = MONO_PostTx / WBC_PostTx,
                perc_NEUT_of_WBC_PostTx = NEUT_PostTx / WBC_PostTx,

                perc_BASO_of_WBC_MidTx = BASO_MidTx / WBC_MidTx,
                perc_EOS_of_WBC_MidTx = EOS_MidTx / WBC_MidTx,
                perc_LYM_of_WBC_MidTx = LYM_MidTx / WBC_MidTx,
                perc_MONO_of_WBC_MidTx = MONO_MidTx / WBC_MidTx,
                perc_NEUT_of_WBC_MidTx = NEUT_MidTx / WBC_MidTx,

                change_in_prop_BASO = case_when(
                    Cohort == "SCLC" ~ perc_BASO_of_WBC_PostTx - perc_BASO_of_WBC_PreTx,
                    Cohort == "TNBC" ~ perc_BASO_of_WBC_MidTx - perc_BASO_of_WBC_PreTx,
                    Cohort == "CRC" ~ perc_BASO_of_WBC_PostTx - perc_BASO_of_WBC_PreTx
                ),
                change_in_prop_EOS = case_when(
                    Cohort == "SCLC" ~ perc_EOS_of_WBC_PostTx - perc_EOS_of_WBC_PreTx,
                    Cohort == "TNBC" ~ perc_EOS_of_WBC_MidTx - perc_EOS_of_WBC_PreTx,
                    Cohort == "CRC" ~ perc_EOS_of_WBC_PostTx - perc_EOS_of_WBC_PreTx
                ),
                change_in_prop_NEUT = case_when(
                    Cohort == "SCLC" ~ perc_NEUT_of_WBC_PostTx - perc_NEUT_of_WBC_PreTx,
                    Cohort == "TNBC" ~ perc_NEUT_of_WBC_MidTx - perc_NEUT_of_WBC_PreTx,
                    Cohort == "CRC" ~ perc_NEUT_of_WBC_PostTx - perc_NEUT_of_WBC_PreTx,
                ),
                change_in_prop_LYM = case_when(
                    Cohort == "SCLC" ~ perc_LYM_of_WBC_PostTx - perc_LYM_of_WBC_PreTx,
                    Cohort == "TNBC" ~ perc_LYM_of_WBC_MidTx - perc_LYM_of_WBC_PreTx,
                    Cohort == "CRC" ~ perc_LYM_of_WBC_PostTx - perc_LYM_of_WBC_PreTx
                ),
                change_in_prop_MONO = case_when(
                    Cohort == "SCLC" ~ perc_MONO_of_WBC_PostTx - perc_MONO_of_WBC_PreTx,
                    Cohort == "TNBC" ~ perc_MONO_of_WBC_MidTx - perc_MONO_of_WBC_PreTx,
                    Cohort == "CRC" ~ perc_MONO_of_WBC_PostTx - perc_MONO_of_WBC_PreTx
                ),
                change_in_prop_GRAN = rowSums(across(c(change_in_prop_BASO, change_in_prop_EOS, change_in_prop_NEUT)), na.rm = TRUE)
            )    

# Adding Blood Count Information from Lab Results to Timepoints
g1_tp <- g1_tp %>%
    left_join(
        g1_lab_changes %>%
        dplyr::select(Cohort, subject, 
            change_in_prop_BASO, change_in_prop_EOS, change_in_prop_LYM, change_in_prop_MONO, change_in_prop_NEUT, change_in_prop_GRAN
        ),
        by = c("DeID"="subject", "Cohort")
    )

# Simplifying Race, Ethnicity Variables due to low sample sizes
g1_tp <- g1_tp %>%
    mutate(
        Race = case_when(
            Race == "White" ~ "White",
            Race == "Unknown" ~ "White",
            TRUE ~ "Non-White"
        ),
        Race = case_when(
            Race == "White" & Ethnicity == "Hispanic or Latino" ~ "Non-White",
            Race == "White" & Ethnicity == "Not Hispanic or Latino" ~ "White",
            Race == "White" & Ethnicity == "Unknown" ~ "White",
            Race == "Non-White" & Ethnicity == "Hispanic or Latino" ~ "Non-White",
            Race == "Non-White" & Ethnicity == "Not Hispanic or Latino" ~ "Non-White",
            Race == "Non-White" & Ethnicity == "Unknown" ~ "Non-White"
        )
    )

# These are the timepoints that we will be using for analysis
table(g1_tp$compare_group, g1_tp$Cohort)

# Even though Timepoints already has Growth Rate, we will calculate it again to ensure consistency
g1_tp %>%
    mutate(
        growth_rate_check = case_when(
            compare_group == "PrevsPost" ~ log(postVAF/preVAF) / change_in_days,
            compare_group == "C1D1vsC5D1" ~ log(postVAF/preVAF) / change_in_days,
            compare_group == "C1D1vsMaintenance" ~ log(postVAF/preVAF) / change_in_days,
            compare_group == "C1D1vsC7D1" ~ log(duringVAF/preVAF) / change_in_days,
            TRUE ~ NA_real_
        )
    ) %>%
    # We will check if the growth rate differs by more than 1e-8
    filter(abs(growth_rate - growth_rate_check) > 1e-8) %>%
    select(deid_key, Cohort, growth_rate, growth_rate_check)
```

# Reference Variables for Statistics
```{r reference_variables, include=TRUE, message = FALSE, warning = FALSE}
g1_tp$Trilaciclib <- factor(g1_tp$Trilaciclib, levels = c("Untreated", "Trilaciclib", "Placebo"))
g1_tp$Gene_Class <- factor(g1_tp$Gene_Class, levels = c("DTA", "DDR"))
g1_tp$Gene <- factor(g1_tp$Gene, levels = c("TP53", "PPM1D", "CHEK2", "DNMT3A", "TET2", "ASXL1", "SF3B1", "SRSF2", "JAK2"))
g1_tp$Cohort <- factor(g1_tp$Cohort, levels = c("Untreated", "SCLC", "TNBC", "CRC"))
g1_tp$Sex <- factor(g1_tp$Sex, levels = c("Male", "Female"))
#g1_tp$Race <- factor(g1_tp$Race, levels = c("White", "Black", "Asian", "American Indian or Alaska Native", "Other", "Unknown"))
g1_tp$Race <- factor(g1_tp$Race, levels = c("White", "Non-White"))
#g1_tp$Ethnicity <- factor(g1_tp$Ethnicity, levels = c("Hispanic or Latino", "Not Hispanic or Latino", "Unknown"))
g1_tp$SmokingStatus <- factor(g1_tp$SmokingStatus, levels = c("Never", "Former", "Current", "Unknown"))
```

# Mutation Table - SCLC and Untreated ONLY
```{r mutation_table, include=TRUE, message = FALSE, warning = FALSE}
num_mutations_per_person <- g1_df %>%
    filter(Cohort == "SCLC" | Cohort == "Untreated") %>%
    filter(average_AF >= 0.002) %>%                             # Anything less than this may be noise.
    group_by(DeID, Gene, Trilaciclib, whichdraw) %>%
    summarise(n = sum(putative_driver)) %>%
    filter(Gene != "") %>%
    #mutate(Gene = ifelse(Gene == "", "No Mutation", Gene)) %>%
    pivot_wider(names_from = Gene, values_from = n, values_fill = 0) %>%
    ungroup() #%>%
    # mutate(
    #     DDR_Mutations = rowSums(.[, c("PPM1D", "TP53", "CHEK2")], na.rm = TRUE),
    # )

g1_mut_table <- g1_demo %>% 
    filter(Cohort == "SCLC" | Cohort == "Untreated") %>%
    dplyr::select(DeID, Trilaciclib, Age, whichdraw) %>% 
    left_join(num_mutations_per_person) %>%
    group_by(DeID, Trilaciclib, Age, whichdraw) %>%
    summarise_if(is.numeric, function(x) sum(x != 0, na.rm = TRUE))

g1_mut_table %>%
    filter(whichdraw == "PreTx") %>%
    mutate(
        CH_Binary = ifelse(rowSums(across(where(is.integer)), na.rm = TRUE) > 0, 1, 0)
    ) %>%
    group_by(Trilaciclib) %>%
    summarise(n_ch = sum(CH_Binary), n_total = n())
```

# Figure 1B - Ribbon Plot
```{r fig1b, fig.width=7.6, fig.height=7}
fig1b <- g1_mut_table %>%
    filter(whichdraw == "PreTx", Age >= 45, Age <= 85) %>%
    mutate(
        CH_Binary = ifelse(rowSums(across(where(is.integer)), na.rm = TRUE) > 0, 1, 0),
        Treatment = ifelse(Trilaciclib == "Untreated", "Healthy Control", "SCLC Pre-Treatment"),
        Treatment = factor(Treatment, levels = c("Healthy Control", "SCLC Pre-Treatment"))
    ) %>%
    ggplot(
        aes(
            x = Age,
            y = CH_Binary,
            fill = Treatment
        )
    ) +
    geom_smooth(
        aes(
            fill = Treatment,
            color = Treatment
        ),
        method = "gam",
        formula = y ~ s(x),
        method.args = list(family = "binomial"),
        size = 1.5,
        se = TRUE,
        alpha = 0.1
    ) +
    scale_y_continuous(labels = scales::label_percent(accuracy = 1L), expand = expansion(add = c(0.01, 0.01))) +
    scale_x_continuous(breaks = seq(0, 90, by = 5)) + 
    labs(x = "Age", y = "Frequency (%)") +
    panel_theme_basic +
    scale_fill_nejm() +
    scale_color_nejm() +
    #scale_fill_manual(values = treatment_colors) +
    #scale_color_manual(values = treatment_colors) +
    theme(
        plot.margin = margin(0, 1, 1, 1, "cm")
    )
fig1b
ggsave(paste0(figures_save_path, "Figure1B.png"), plot = fig1b, width = 14, height = 8, dpi = 300)
```

# Figure 1C - Waterfall Plot
```{r fig1c}
df_to_maf <- g1_df %>% filter(Cohort == "SCLC" | Cohort == "Untreated") %>%
    filter(VariantClass != "synonymous_variant", putative_driver == 1, whichdraw == "PreTx") %>%
    tidyr::separate(key, into= c("Chromosome", "Start_Position", "Reference_Allele", "Tumor_Seq_Allele2"), sep = ":") %>%
    mutate(
        Tumor_Sample_Barcode = DeID,
        Hugo_Symbol = Gene,
        Variant_Classification = VariantClass,
        Start_Position = as.numeric(Start_Position),
        Variant_Type = case_when(
            nchar(Reference_Allele) == nchar(Tumor_Seq_Allele2) ~ "SNP",
            nchar(Reference_Allele) < nchar(Tumor_Seq_Allele2) ~ "INS",
            nchar(Reference_Allele) > nchar(Tumor_Seq_Allele2) ~ "DEL"
        ),
        End_Position = case_when(
            Variant_Type == "SNP" ~ Start_Position,
            Variant_Type == "INS" ~ Start_Position,
            Variant_Type == "DEL" ~ (Start_Position + nchar(Reference_Allele) - 1)
        ),
        stringsAsFactors = FALSE
    )
#fwrite(df_to_maf, paste0(data_file_path, "waterfall_plots_raw_data.csv"), row.names = FALSE)

nejmPal <- pal_nejm("default")(8)
col <- nejmPal[1:6]
names(col) <- c("Multi mutation", "Frameshift", "Missense", "Nonsense", "Splicing", "Other")

# Function to define colors for alteration cells
alter_fun_vec <- function(x, y, w, h, v) {
    tmp <- pal_nejm("default")(6)
    names(tmp) <- c("Multi mutation", "Frameshift", "Missense", "Nonsense", "Splicing", "Other")
    colNames <- names(v)[!is.na(names(v))]
    v <- matrix(v, ncol = length(colNames))
    color <- apply(v, 1, function(x) {
        if (sum(x) == 0) {
            "#FFFFFF"
        } else {
            tmp[colNames[x]][1]
        }
    })
    grid.rect(x, y, w * 1.1, h,
        gp = gpar(fill = color, col = NA)
    )
}

# Preparing Data for OncoPrint
toPlot <- df_to_maf %>%
    dplyr::select(DeID, Gene, VariantClass, Cohort, Trilaciclib) %>%
    dplyr::filter(Gene != "") %>%
    dplyr::filter(Cohort %in% c("Untreated", "SCLC")) %>%
    group_by(DeID, Gene) %>%
    mutate(VariantClass = ifelse(n() > 1, "multi_mutation", VariantClass)) %>%
    ungroup() %>%
    distinct() %>%
    mutate(
        Cohort = case_match(
            Cohort,
            "Untreated" ~ "Healthy Control",
            "SCLC" ~ "SCLC Pre-Treatment"
        ),
        VariantClass = case_match(
            VariantClass,
            "multi_mutation" ~ "Multi mutation",
            "frameshift_variant" ~ "Frameshift",
            c("missense_variant", "inframe_deletion", "inframe_insertion") ~ "Missense",
            "splice_region_variant" ~ "Splicing",
            c("start_lost", "stop_gained") ~ "Nonsense",
            .default = "Other"
        )
    )

# Creating Matrix for OncoPrint
mat <- toPlot %>%
    dplyr::select(DeID, Gene, VariantClass) %>%
    pivot_wider(names_from = DeID, values_from = VariantClass, values_fill = "") %>%
    column_to_rownames("Gene")
# Ordering rows by number of mutations per gene
rowOrder <- toPlot %>%
    group_by(Gene) %>%
    summarise(n = n()) %>%
    arrange(desc(n)) %>%
    pull(Gene) %>%
    unique()
# Ordering columns by Cohort, Gene, VariantClass, and number of mutations
colOrder <- toPlot %>%
    group_by(DeID) %>%
    mutate(n = sum(VariantClass != "")) %>%
    ungroup() %>%
    arrange(
        factor(Cohort, levels = c("Healthy Control", "SCLC Pre-Treatment")),
        factor(Gene, levels = rowOrder),
        factor(VariantClass, levels = c("Multi mutation", "Frameshift", "Missense", "Nonsense", "Splicing", "Other")),
        desc(n)
    ) %>%
    pull(DeID) %>%
    unique()
# Reordering matrix
mat <- mat[rowOrder, colOrder]
# Creating Column Information for Annotations
colInfor <- toPlot %>%
    dplyr::select(DeID, Cohort, Trilaciclib) %>%
    distinct() %>%
    arrange(factor(DeID, levels = colOrder))

p <- oncoPrint(mat,
    alter_fun = alter_fun_vec,
    alter_fun_is_vectorized = TRUE,
    col = col,
    row_order = rownames(mat),
    column_order = colnames(mat),
    show_pct = TRUE,
    pct_side = "right",
    row_names_side = "left",
    top_annotation = HeatmapAnnotation(
        annotation_name_gp = gpar(fontsize = 10),
        annotation_name_side = "left",
        annotation_name_rot = 90,
        annotation_name_offset = unit(15, "points"),
        "# of Muts" = anno_oncoprint_barplot() %>%
            {
                x <- .
                x@show_name <- TRUE
                x
            }
    ),
    right_annotation = rowAnnotation(
        annotation_name_gp = gpar(fontsize = 10),
        annotation_name_side = "top",
        annotation_name_offset = unit(15, "points"),
        "# of Samples" = anno_oncoprint_barplot() %>%
            {
                x <- .
                x@show_name <- TRUE
                x
            }
    ),
    bottom_annotation = HeatmapAnnotation(
        Cohort = colInfor$Cohort,
        Treatment = colInfor$Trilaciclib,
        col = list(
            Cohort = c(`Healthy Control` = nejmPal[1], `SCLC Pre-Treatment` = nejmPal[2]),
            Treatment = c(Untreated = nejmPal[1], Trilaciclib = nejmPal[2], Placebo = nejmPal[3])
        ),
        annotation_legend_param = list(
            Cohort = list(
                nrow = 1,
                at = c("Healthy Control", "SCLC Pre-Treatment")
            ),
            Treatment = list(
                nrow = 2,
                at = c("Untreated", "Trilaciclib", "Placebo"),
                by_row = TRUE
            )
        )
    ),
    heatmap_legend_param = list(
        title = "Alteration",
        nrow = 2,
        by_row = TRUE,
        at = c("Multi mutation", "Frameshift", "Missense", "Nonsense", "Splicing", "Other")
    )
)

png(paste0(figures_save_path, "Figure1C.png"), width = 8, height = 5, units = "in", res = 300)
draw(
    p,
    heatmap_legend_side = "bottom",
    annotation_legend_side = "bottom",
    merge_legend = TRUE,
    padding = unit(c(5.5, 5.5, 5.5, 10), "points")
)
dev.off()
```

# Figure 1D - VAF Bin Mutation Bar Plot
```{r fig1d, fig.width=7, fig.height=7.6}
df_to_plot <- g1_df %>% filter(Cohort == "SCLC" | Cohort == "Untreated") %>%
    filter(whichdraw == "PreTx") %>%
    mutate(
        vaf_bin = case_when(
            average_AF >= 0.002 & average_AF < 0.01 ~ "0.2-1% VAF", 
            average_AF >= 0.01 & average_AF < 0.05 ~ "1-5% VAF",
            average_AF >= 0.05 ~ ">=5% VAF",
            average_AF >= 0.001 & average_AF < 0.002 ~ "0.1-0.2% VAF"
        )
    ) %>%
    filter(vaf_bin != "<0.2% VAF") %>%
    group_by(Gene, vaf_bin) %>%
    summarise(n = n()) %>%
    group_by(Gene) %>%
    mutate(
        total = sum(n),
        percentage = (n / total) * 100,
        vaf_bin = factor(vaf_bin, levels = c(">=5% VAF", "1-5% VAF", "0.2-1% VAF", "0.1-0.2% VAF")),
        Gene = factor(Gene, levels = c(gene_list))
    )

fig1d <- df_to_plot %>%
    ggplot(
        aes(
            x = Gene,
            y = percentage,
            fill = vaf_bin
        )
    ) +
    geom_bar(stat = "identity") +
    labs(
        x = "Gene",
        y = "Percentage of Mutations at Baseline",
        fill = "VAF Bin"
    ) +
    panel_theme_basic +
    theme(
        axis.text.x = element_text(angle = 45, hjust = 1),
        plot.margin = margin(2, 1, 1, 1, "cm")
    ) +
    scale_fill_nejm()
fig1d
ggsave(paste0(figures_save_path, "Figure1D.png"), plot = fig1d, width = 10, height = 8, dpi = 300)
```

# Figure 1E - Proportion of Patients with CH Mutations
```{r fig1e}
model_treatment_gene <- map_dfr(c(gene_list), function(gene){
    rbind(
        glm(
            formula = as.formula(paste(gene, "~ Trilaciclib + Age")),
            family = binomial(link = "logit"),
            data = g1_mut_table %>%
                filter(whichdraw == "PostTx") %>%
                mutate(Trilaciclib = factor(Trilaciclib, levels = c("Untreated", "Placebo", "Trilaciclib")))
        ) %>%
        sjPlot::get_model_data(type = "est") %>%
        mutate(Gene = gene),
        glm(
            formula = as.formula(paste(gene, "~ Trilaciclib + Age")),
            family = binomial(link = "logit"),
            data = g1_mut_table %>%
                filter(whichdraw == "PostTx") %>%
                filter(Trilaciclib != "Untreated") %>%
                mutate(Trilaciclib = factor(Trilaciclib, levels = c("Trilaciclib", "Placebo")))
        ) %>%
        sjPlot::get_model_data(type = "est") %>%
        mutate(Gene = gene)
    )
})
model_treatment_gene <- model_treatment_gene %>% filter(term != "Age")

g1_mut_table <- g1_mut_table %>%
    mutate(
        DTA_Mutations = ifelse(rowSums(across(c("DNMT3A", "TET2", "ASXL1")), na.rm = TRUE) > 0, 1, 0),
        DDR_Mutations = ifelse(rowSums(across(c("PPM1D", "TP53", "CHEK2")), na.rm = TRUE) > 0, 1, 0),
        Treatment = ifelse(Trilaciclib == "Untreated", "Healthy Control", "SCLC Post-Treatment"),
        Treatment = factor(Treatment, levels = c("Healthy Control", "SCLC Post-Treatment"))
    )

g1_mut_table %>%
    filter(whichdraw == "PostTx") %>%
    group_by(Trilaciclib) %>%
    summarise(n_ch = sum(DDR_Mutations), n_total = n())

fig1e <- g1_mut_table %>%
    filter(whichdraw == "PostTx") %>%
    group_by(Treatment) %>%
    dplyr::select(-Age, -Trilaciclib, -whichdraw, -DTA_Mutations, -DDR_Mutations) %>%
    summarise_if(is.numeric, sum, na.rm = TRUE) %>%
    pivot_longer(cols = -c(Treatment), names_to = "Gene", values_to = "n") %>%
    group_by(Treatment, Gene) %>%
    left_join(
        g1_mut_table %>%
            filter(whichdraw == "PostTx") %>%
            group_by(Treatment) %>%
            summarise(total_patients = n())
    ) %>%
    mutate(
        Proportion = n / total_patients,
        Gene = factor(Gene, levels = c(gene_list)),
        Treatment = factor(Treatment, levels = c("Healthy Control", "SCLC Post-Treatment"))
        #Trilaciclib = factor(Trilaciclib, levels = c("Untreated", "Placebo", "Trilaciclib"))
    ) %>%
    ggplot(
        aes(
        x = Gene,
        y = Proportion,
        fill = Treatment
        )
    ) +
    geom_bar(color = "black", stat = "identity", position = position_dodge()) +
    labs(
        x = "",
        y = "Percent of Patients\nwith Mutated Gene\n Post-Treatment",
        fill = "Treatment"
    ) +
    panel_theme_basic +
    theme(
        axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = c(0.7, 0.8),
        legend.direction = "vertical",
        plot.margin = margin(2, 0, 1, 1, "cm")
    ) +
    scale_y_continuous(labels = scales::percent, limits = c(NA, 0.6), expand = c(0, 0)) +
    scale_fill_nejm() +
    scale_color_nejm()

# If we want to add significance annotations, we can use the following code
    # do.call(geom_signif, default_sig(0.6, 0.75, 1, model_treatment_gene$p.value[1], model_treatment_gene$p.stars[1], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.65, 0.75, 1.25, model_treatment_gene$p.value[2], model_treatment_gene$p.stars[2], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.7, 1, 1.25, model_treatment_gene$p.value[3], model_treatment_gene$p.stars[3], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.35, 1.75, 2, model_treatment_gene$p.value[4], model_treatment_gene$p.stars[4], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.4, 1.75, 2.25, model_treatment_gene$p.value[5], model_treatment_gene$p.stars[5], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.45, 2, 2.25, model_treatment_gene$p.value[6], model_treatment_gene$p.stars[6], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.2, 2.75, 3, model_treatment_gene$p.value[7], model_treatment_gene$p.stars[7], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.25, 2.75, 3.25, model_treatment_gene$p.value[8], model_treatment_gene$p.stars[8], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.3, 3, 3.25, model_treatment_gene$p.value[9], model_treatment_gene$p.stars[9], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.45, 3.75, 4, model_treatment_gene$p.value[10], model_treatment_gene$p.stars[10], show_values = FALSE, hide_ns = FALSE)) +
    # do.call(geom_signif, default_sig(0.52, 3.75, 4.25, model_treatment_gene$p.value[11], model_treatment_gene$p.stars[11], show_values = FALSE, hide_ns = FALSE)) +
    # do.call(geom_signif, default_sig(0.6, 4, 4.25, model_treatment_gene$p.value[12], model_treatment_gene$p.stars[12], show_values = FALSE, hide_ns = FALSE)) +
    # do.call(geom_signif, default_sig(0.45, 4.75, 5, model_treatment_gene$p.value[13], model_treatment_gene$p.stars[13], show_values = FALSE, hide_ns = FALSE)) +
    # do.call(geom_signif, default_sig(0.52, 4.75, 5.25, model_treatment_gene$p.value[14], model_treatment_gene$p.stars[14], show_values = FALSE, hide_ns = FALSE)) +
    # do.call(geom_signif, default_sig(0.6, 5, 5.25, model_treatment_gene$p.value[15], model_treatment_gene$p.stars[15], show_values = FALSE, hide_ns = FALSE)) +
    # do.call(geom_signif, default_sig(0.25, 5.75, 6, model_treatment_gene$p.value[16], model_treatment_gene$p.stars[16], show_values = FALSE, hide_ns = FALSE)) +
    # do.call(geom_signif, default_sig(0.32, 5.75, 6.25, model_treatment_gene$p.value[17], model_treatment_gene$p.stars[17], show_values = FALSE, hide_ns = FALSE)) +
    # do.call(geom_signif, default_sig(0.4, 6, 6.25, model_treatment_gene$p.value[18], model_treatment_gene$p.stars[18], show_values = FALSE, hide_ns = FALSE)) +
    # do.call(geom_signif, default_sig(0.1, 6.75, 7, model_treatment_gene$p.value[19], model_treatment_gene$p.stars[19], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.15, 6.75, 7.25, model_treatment_gene$p.value[20], model_treatment_gene$p.stars[20], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.2, 7, 7.25, model_treatment_gene$p.value[21], model_treatment_gene$p.stars[21], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.1, 7.75, 8, model_treatment_gene$p.value[22], model_treatment_gene$p.stars[22], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.15, 7.75, 8.25, model_treatment_gene$p.value[23], model_treatment_gene$p.stars[23], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.2, 8, 8.25, model_treatment_gene$p.value[24], model_treatment_gene$p.stars[24], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.1, 8.75, 9, model_treatment_gene$p.value[25], model_treatment_gene$p.stars[25], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.15, 8.75, 9.25, model_treatment_gene$p.value[26], model_treatment_gene$p.stars[26], show_values = FALSE)) +
    # do.call(geom_signif, default_sig(0.2, 9, 9.25, model_treatment_gene$p.value[27], model_treatment_gene$p.stars[27], show_values = FALSE)) #+
    #do.call(geom_signif, default_sig(0.6, 9.75, 10, model_treatment_gene$p.value[28], model_treatment_gene$p.stars[28], show_values = FALSE, hide_ns = FALSE)) +
    #do.call(geom_signif, default_sig(0.65, 9.75, 10.25, model_treatment_gene$p.value[29], model_treatment_gene$p.stars[29], show_values = FALSE, hide_ns = FALSE)) +
    #do.call(geom_signif, default_sig(0.7, 10, 10.25, model_treatment_gene$p.value[30], model_treatment_gene$p.stars[30], show_values = FALSE, hide_ns = FALSE)) 

fig1e
ggsave(paste0(figures_save_path, "Figure1E.png"), plot = fig1e, width = 10, height = 6, dpi = 300)
```

# Figure 1E - Proportion of Patients with CH Mutations Continued...
```{r fig1e_continued, fig.width=10, fig.height=10}
model_treatment_gene_class <- map_dfr(c("DTA_Mutations", "DDR_Mutations"), function(gene_class){
    rbind(
        glm(
            formula = as.formula(paste(gene_class, "~ Trilaciclib + Age")),
            family = binomial(link = "logit"),
            data = g1_mut_table %>%
                filter(whichdraw == "PostTx") %>%
                filter(Trilaciclib != "Untreated") %>%
                mutate(Trilaciclib = factor(Trilaciclib, levels = c("Trilaciclib", "Placebo")))
        ) %>%
        sjPlot::get_model_data(type = "est") %>%
        mutate(
            GeneClass = ifelse(gene_class == "DTA_Mutations", "DTA", "DDR"))
    )
})
model_treatment_gene_class <- model_treatment_gene_class %>% filter(term != "Age")

fig1e_cont <- g1_mut_table %>%
    filter(whichdraw == "PostTx", Trilaciclib != "Untreated") %>%
    group_by(Trilaciclib) %>%
    dplyr::select(Trilaciclib, DTA_Mutations, DDR_Mutations) %>%
    summarise_if(is.numeric, sum, na.rm = TRUE) %>%
    pivot_longer(cols = -c(Trilaciclib), names_to = "Gene", values_to = "n") %>%
    left_join(
        g1_mut_table %>%
            filter(whichdraw == "PostTx", Trilaciclib != "Untreated") %>%
            group_by(Trilaciclib) %>%
            summarise(total_patients = n())
    ) %>%
    mutate(
        Proportion = n / total_patients,
        Trilaciclib = factor(Trilaciclib, levels = c("Placebo", "Trilaciclib")),
        Gene = ifelse(Gene == "DTA_Mutations", "DTA", "DDR"),
        Gene = factor(Gene, levels = c("DTA", "DDR"))
    ) %>%
    ggplot(
        aes(
        x = Gene,
        y = Proportion,
        fill = Trilaciclib
        )
    ) +
    geom_bar(color = "black", stat = "identity", position = position_dodge()) +
    labs(
        x = "",
        y = "Percent of Patients\nwith Mutated Gene\nPost-Treatment",
        fill = "Trilaciclib"
    ) +
    panel_theme_basic +
    theme(
        axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = c(0.85, 0.6),
        legend.direction = "vertical",
        #plot.margin = margin(1, 0, 1, 1, "cm")
    ) +
    scale_y_continuous(labels = scales::percent, limits = c(NA, 0.75), expand = c(0, 0)) +
    scale_fill_manual(values = treatment_colors) +
    do.call(geom_signif, default_sig(0.65, 0.75, 1.25, model_treatment_gene_class$p.value[1], model_treatment_gene_class$p.stars[1], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig(0.65, 1.75, 2.25, model_treatment_gene_class$p.value[2], model_treatment_gene_class$p.stars[2], show_values = FALSE, hide_ns = FALSE))
fig1e_cont
ggsave(paste0(figures_save_path, "Figure1E_Continued.png"), plot = fig1e_cont, width = 10, height = 6, dpi = 300)
```

# Figure 1F - Small Cell Lung Cancer Growth Rate Plot
```{r fig1f, fig.width=4.8, fig.height=4.2}
df_to_plot <- g1_tp %>%
    filter(Cohort == "SCLC" | Cohort == "Untreated") %>%    # Filter for SCLC and Untreated cohorts
    filter(Gene_Class == "DTA" | Gene_Class == "DDR") %>%   # Filter for DTA and DDR gene classes
    filter(growth_rate != "Inf" & growth_rate != "-Inf")    # Remove Inf and -Inf growth rates

model_growth_rate_class <- model_growth_rate_class_fun(df_to_plot)

# Absolute DDR Growth Rate Calculations
df_to_plot %>%
    filter(Gene_Class == "DDR") %>%
    group_by(Trilaciclib) %>%
    summarise(
        median_gr = median(growth_rate),
        average_gr = mean(growth_rate),
    ) %>%
    pivot_wider(names_from = Trilaciclib, values_from = c(median_gr, average_gr)) %>%
    mutate(perc_reduction = (1 - median_gr_Trilaciclib / median_gr_Placebo) * 100) %>%
    select(perc_reduction)

counts <- df_to_plot %>%
    group_by(Trilaciclib, Gene_Class) %>%
    summarise(n = n(), .groups = "drop")

fig1f <- df_to_plot %>%
    ggplot(
        aes(
            x = Gene_Class,                 # Notice how we are using the Gene_Class variable for the x-axis where before we were using Trilaciclib.
            y = growth_rate, 
            fill = Trilaciclib              # We will still group data by Trilaciclib, but our main focus is on the Gene_Class variable.
        )
    ) +
    geom_hline(yintercept = 0, color = "black", alpha = 0.3) +
    geom_boxplot(outlier.shape = NA) +
    geom_point(position = position_jitterdodge(jitter.width=0.2), alpha = 0.3) +
    geom_text(data = counts, aes(label = n, y = 0.06), size = 8, position = position_dodge(width = 0.8)) +
    labs(
        y = "Growth Rate", 
        x = '', 
        title = ""
    ) +
    panel_theme_basic +
    scale_y_continuous(trans = scales::pseudo_log_trans(0.001), breaks = c(-0.05, -0.01, -0.005, 0, 0.005, 0.01, 0.05), limit = c(NA, 1.5)) +
    scale_fill_manual(values = treatment_colors) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 5, 1.75, 2, model_growth_rate_class$p.value[3], model_growth_rate_class$p.stars[3], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 6, 1.75, 2.25, model_growth_rate_class$p.value[4], model_growth_rate_class$p.stars[4], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 7, 2, 2.25, model_growth_rate_class$p.value[6], model_growth_rate_class$p.stars[6], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 5, 0.75, 1, model_growth_rate_class$p.value[1], model_growth_rate_class$p.stars[1], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 6, 0.75, 1.25, model_growth_rate_class$p.value[2], model_growth_rate_class$p.stars[2], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 7, 1, 1.25, model_growth_rate_class$p.value[5], model_growth_rate_class$p.stars[5], show_values = FALSE))
fig1f
ggsave(paste0(figures_save_path, "Figure1F.png"), plot = fig1f, width = 12, height = 10, dpi = 300)
```

# Figure 2C - Colorectal Cancer Growth Rate Plot
```{r fig2c, fig.width=10, fig.height=10}
df_to_plot <- g1_tp %>% 
    filter(Cohort == "CRC" | Cohort == "Untreated") %>%     # Filter for CRC and Untreated cohorts
    filter(Gene_Class == "DTA" | Gene_Class == "DDR") %>%   # Filter for DTA and DDR gene classes
    filter(growth_rate != "Inf" & growth_rate != "-Inf")    # Remove Inf and -Inf growth rates

model_growth_rate_class <- model_growth_rate_class_fun(df_to_plot)

# Absolute DDR Growth Rate Calculations
df_to_plot %>%
    filter(Gene_Class == "DDR") %>%
    group_by(Trilaciclib) %>%
    summarise(
        median_gr = median(growth_rate),
        average_gr = mean(growth_rate),
    ) %>%
    pivot_wider(names_from = Trilaciclib, values_from = c(median_gr, average_gr)) %>%
    mutate(perc_reduction = (1 - median_gr_Trilaciclib / median_gr_Placebo) * 100) %>%
    select(perc_reduction)

counts <- df_to_plot %>%
    group_by(Trilaciclib, Gene_Class) %>%
    summarise(n = n(), .groups = "drop")

fig2c <- df_to_plot %>%
    filter(growth_rate != "Inf" & growth_rate != "-Inf") %>%
    filter(Gene_Class == "DDR" | Gene_Class == "DTA") %>%       # We are filtering the data to only include the DDR and DTA gene classes.
    ggplot(
        aes(
            x = Gene_Class,                 # Notice how we are using the Gene_Class variable for the x-axis where before we were using Trilaciclib.
            y = growth_rate, 
            fill = Trilaciclib              # We will still group data by Trilaciclib, but our main focus is on the Gene_Class variable.
        )
    ) +
    geom_hline(yintercept = 0, color = "black", alpha = 0.3) +
    geom_boxplot(outlier.shape = NA) +
    geom_point(position = position_jitterdodge(jitter.width=0.2), alpha = 0.3) +
    geom_text(data = counts, aes(label = n, y = 0.06), size = 8, position = position_dodge(width = 0.8)) +
    labs(
        y = "Growth Rate", 
        x = '', 
        title = ""
    ) +
    panel_theme_basic +
    scale_y_continuous(trans = scales::pseudo_log_trans(0.001), breaks = c(-0.05, -0.01, -0.005, 0, 0.005, 0.01, 0.05), limit = c(NA, 1.5)) +
    scale_fill_manual(values = treatment_colors) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 5, 0.75, 1, model_growth_rate_class$p.value[1], model_growth_rate_class$p.stars[1], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 6, 0.75, 1.25, model_growth_rate_class$p.value[2], model_growth_rate_class$p.stars[2], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 7, 1, 1.25, model_growth_rate_class$p.value[5], model_growth_rate_class$p.stars[5], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 5, 1.75, 2, model_growth_rate_class$p.value[3], model_growth_rate_class$p.stars[3], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 6, 1.75, 2.25, model_growth_rate_class$p.value[4], model_growth_rate_class$p.stars[4], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 7, 2, 2.25, model_growth_rate_class$p.value[6], model_growth_rate_class$p.stars[6], show_values = FALSE))
fig2c
ggsave(paste0(figures_save_path, "Figure2C.png"), plot = fig2c, width = 12, height = 8, dpi = 300)
```

# Figure 2D - Triple Negative Breast Cancer Growth Rate Plot
```{r fig2d, fig.width=10, fig.height=10}
df_to_plot <- g1_tp %>%
    filter(Cohort == "TNBC" | Cohort == "Untreated") %>%    # Filter for TNBC and Untreated cohorts
    filter(Gene_Class == "DTA" | Gene_Class == "DDR") %>%   # Filter for DTA and DDR gene classes
    filter(growth_rate != "Inf" & growth_rate != "-Inf")    # Remove Inf and -Inf growth rates

model_growth_rate_class <- model_growth_rate_class_fun(df_to_plot)

df_to_plot %>%
    filter(Gene_Class == "DDR") %>%
    group_by(Trilaciclib) %>%
    summarise(
        median_gr = median(growth_rate),
        average_gr = mean(growth_rate),
    ) %>%
    pivot_wider(names_from = Trilaciclib, values_from = c(median_gr, average_gr)) %>%
    mutate(perc_reduction = (1 - median_gr_Trilaciclib / median_gr_Placebo) * 100) %>%
    select(perc_reduction)

counts <- df_to_plot %>%
    group_by(Trilaciclib, Gene_Class) %>%
    summarise(n = n(), .groups = "drop")

fig2d <- df_to_plot %>%
    ggplot(
        aes(
            x = Gene_Class,                 # Notice how we are using the Gene_Class variable for the x-axis where before we were using Trilaciclib.
            y = growth_rate, 
            fill = Trilaciclib              # We will still group data by Trilaciclib, but our main focus is on the Gene_Class variable.
        )
    ) +
    geom_hline(yintercept = 0, color = "black", alpha = 0.3) +
    geom_boxplot(outlier.shape = NA) +
    geom_point(position = position_jitterdodge(jitter.width=0.2), alpha = 0.3) +
    geom_text(data = counts, aes(label = n, y = 0.06), size = 8, position = position_dodge(width = 0.8)) +
    labs(
        y = "Growth Rate", 
        x = '', 
        title = ""
    ) +
    panel_theme_basic +
    scale_y_continuous(trans = scales::pseudo_log_trans(0.001), breaks = c(-0.05, -0.01, -0.005, 0, 0.005, 0.01, 0.05), limit = c(NA, 1.5)) +
    scale_fill_manual(values = treatment_colors) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 5, 1.75, 2, model_growth_rate_class$p.value[3], model_growth_rate_class$p.stars[3], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 6, 1.75, 2.25, model_growth_rate_class$p.value[4], model_growth_rate_class$p.stars[4], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 7, 2, 2.25, model_growth_rate_class$p.value[6], model_growth_rate_class$p.stars[6], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 5, 0.75, 1, model_growth_rate_class$p.value[1], model_growth_rate_class$p.stars[1], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 6, 0.75, 1.25, model_growth_rate_class$p.value[2], model_growth_rate_class$p.stars[2], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 7, 1, 1.25, model_growth_rate_class$p.value[5], model_growth_rate_class$p.stars[5], show_values = FALSE))
fig2d
ggsave(paste0(figures_save_path, "Figure2D.png"), plot = fig2d, width = 12, height = 8, dpi = 300)
```

# Figure 2E - Genes
```{r fig2e, fig.width=10, fig.height=10}
df_to_plot <- g1_tp %>% 
    filter(Gene_Class == "DDR") %>%                             # Filter for DDR gene class
    filter(growth_rate != "Inf" & growth_rate != "-Inf") %>%    # Remove Inf and -Inf growth rates
    mutate(Variant_Classification = if_else(VariantClass == "missense_variant", "Missense", "Nonsense"))

model_growth_rate_gene <- purrr::cross_df(list(
    gene = c("TP53", "PPM1D", "CHEK2"),
    comparison = c("all", "no_untreated")
)) %>%
    pmap_dfr(function(gene, comparison) {
        print(paste(gene, comparison))
        if (comparison == "all") {
            df_subset <- df_to_plot %>% filter(Gene == gene)
            model <- glm(
                formula = growth_rate ~ Trilaciclib + Age,
                data = df_subset
            ) %>%
                sjPlot::get_model_data(type = "est") %>%
                filter(!grepl("Age", term), !grepl("Cohort", term)) %>%
                mutate(term = ifelse(term == "TrilaciclibTrilaciclib", "UntreatedVSTrilaciclib", "UntreatedVSPlacebo"))
        } else if (comparison == "no_untreated") {
            df_subset <- df_to_plot %>% filter(Gene == gene & Trilaciclib != "Untreated")
            model <- glm(
                formula = growth_rate ~ Trilaciclib + Cohort + Age,
                data = df_subset
            ) %>%
                sjPlot::get_model_data(type = "est") %>%
                filter(!grepl("Age", term), !grepl("Cohort", term)) %>%
                mutate(term = "TrilaciclibVSPlacebo")
        }
        model %>% mutate(gene = gene, comparison = comparison)
    })

glm(
    formula = growth_rate ~ Trilaciclib + Age + Cohort,
    data = df_to_plot %>% filter(Gene == "TP53", Trilaciclib != "Untreated")
) %>%
    sjPlot::get_model_data(type = "est")

model_growth_rate_variant_class <- purrr::cross_df(list(
    classification = c("Missense", "Nonsense"),
    comparison = c("all", "no_untreated")
)) %>%
    pmap_dfr(function(classification, comparison) {
        print(paste(classification, comparison))
        if (comparison == "all") {
            df_subset <- df_to_plot %>% filter(Gene == "TP53", Variant_Classification == classification)
            model <- glm(
                formula = growth_rate ~ Trilaciclib + Age,
                data = df_subset
            ) %>%
                sjPlot::get_model_data(type = "est") %>%
                filter(!grepl("Age", term), !grepl("Cohort", term)) %>%
                mutate(term = ifelse(term == "TrilaciclibTrilaciclib", "UntreatedVSTrilaciclib", "UntreatedVSPlacebo"))
        } else if (comparison == "no_untreated") {
            df_subset <- df_to_plot %>% filter(Gene == "TP53", Variant_Classification == classification, Trilaciclib != "Untreated")
            model <- glm(
                formula = growth_rate ~ Trilaciclib + Cohort + Age,
                data = df_subset
            ) %>%
                sjPlot::get_model_data(type = "est") %>%
                filter(!grepl("Age", term), !grepl("Cohort", term)) %>%
                mutate(term = "TrilaciclibVSPlacebo")
        }
        model %>% mutate(classification = classification, comparison = comparison)
    })

df_to_plot %>% dplyr::select(Gene, Trilaciclib) %>% table()

counts_genes <- df_to_plot %>%
    group_by(Trilaciclib, Gene) %>%
    summarise(n = n(), .groups = "drop")

counts_variant_class <- df_to_plot %>%
    filter(Gene == "TP53") %>%
    group_by(Trilaciclib, Variant_Classification) %>%
    summarise(n = n(), .groups = "drop")

fig2e <- df_to_plot %>%
    ggplot(
        aes(
            x = Gene,
            y = growth_rate, 
            fill = Trilaciclib
        )
    ) +
    geom_hline(yintercept = 0, color = "black", alpha = 0.3) +
    geom_boxplot(outlier.shape = NA) +
    geom_point(position = position_jitterdodge(jitter.width=0.2), alpha = 0.3) +
    geom_text(data = counts_genes, aes(label = n, y = 0.06), size = 6, position = position_dodge(width = 0.8)) +
    labs(
        y = "Growth Rate", 
        x = '', 
        title = ""
    ) +
    panel_theme_basic +
    theme(
        legend.text = element_text(size = 18),
        axis.text.y = element_text(size = 18),
        axis.text.x = element_text(size = 18),
        axis.title.y = element_text(size = 18),
        axis.title.x = element_text(size = 18),
    ) +
    scale_y_continuous(trans = scales::pseudo_log_trans(0.001), breaks = c(-0.05, -0.01, -0.005, 0, 0.005, 0.01, 0.05), limit = c(NA, 1.5)) +
    scale_fill_manual(values = treatment_colors) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "TP53"), 5, 0.75, 1, model_growth_rate_gene$p.value[1], model_growth_rate_gene$p.stars[1], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "TP53"), 6, 0.75, 1.25, model_growth_rate_gene$p.value[2], model_growth_rate_gene$p.stars[2], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "TP53"), 7, 1, 1.25, model_growth_rate_gene$p.value[7], model_growth_rate_gene$p.stars[7], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "PPM1D"), 5, 1.75, 2, model_growth_rate_gene$p.value[3], model_growth_rate_gene$p.stars[3], show_values = FALSE, hide_ns = FALSE)) + 
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "PPM1D"), 6, 1.75, 2.25, model_growth_rate_gene$p.value[4], model_growth_rate_gene$p.stars[4], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "PPM1D"), 7.1, 2, 2.25, model_growth_rate_gene$p.value[8], model_growth_rate_gene$p.stars[8], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "CHEK2"), 5, 2.75, 3, model_growth_rate_gene$p.value[5], model_growth_rate_gene$p.stars[5], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "CHEK2"), 6, 2.75, 3.25, model_growth_rate_gene$p.value[6], model_growth_rate_gene$p.stars[6], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "CHEK2"), 7.1, 3, 3.25, model_growth_rate_gene$p.value[9], model_growth_rate_gene$p.stars[9], show_values = FALSE, hide_ns = FALSE)) 

ext_fig_tp53 <- df_to_plot %>%
    filter(Gene == "TP53") %>%
    ggplot(
        aes(
            x = Variant_Classification,
            y = growth_rate, 
            fill = Trilaciclib
        )
    ) +
    geom_hline(yintercept = 0, color = "black", alpha = 0.3) +
    geom_boxplot(outlier.shape = NA) +
    geom_point(position = position_jitterdodge(jitter.width=0.2), alpha = 0.3) +
    geom_text(data = counts_variant_class, aes(label = n, y = 0.06), size = 8, position = position_dodge(width = 0.8)) +
    labs(
        y = "Growth Rate", 
        x = 'Missense', 
        title = "TP53"
    ) +
    panel_theme_basic +
    scale_y_continuous(trans = scales::pseudo_log_trans(0.001), breaks = c(-0.05, -0.01, -0.005, 0, 0.005, 0.01, 0.05), limit = c(NA, 1.5)) +
    scale_fill_manual(values = treatment_colors) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "TP53"), 5, 0.75, 1, model_growth_rate_variant_class$p.value[1], model_growth_rate_variant_class$p.stars[1], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "TP53"), 6, 0.75, 1.25, model_growth_rate_variant_class$p.value[2], model_growth_rate_variant_class$p.stars[2], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "TP53"), 7, 1, 1.25, model_growth_rate_variant_class$p.value[5], model_growth_rate_variant_class$p.stars[5], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "TP53"), 5, 1.75, 2, model_growth_rate_variant_class$p.value[3], model_growth_rate_variant_class$p.stars[3], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "TP53"), 6, 1.75, 2.25, model_growth_rate_variant_class$p.value[4], model_growth_rate_variant_class$p.stars[4], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "TP53"), 7, 2, 2.25, model_growth_rate_variant_class$p.value[6], model_growth_rate_variant_class$p.stars[6], show_values = FALSE, hide_ns = FALSE))

fig2e
ext_fig_tp53

ggsave(paste0(figures_save_path, "Figure2E.png"), plot = fig2e, width = 20, height = 8, dpi = 300)
```

# Extended Figure - These take a long time to run and sometimes the server crashes, so they are commented out.
```{r lollipop, include=FALSE}
colors.mutation_type <- brewer.pal(8,"Dark2")[1:8]
names(colors.mutation_type) <- c("frameshift_variant", "inframe_insertion", "splice_region_variant", "inframe_deletion", 'start_lost','stop_gained','missense_variant', "synonymous_variant")

prot.info = get_protein_info("TP53")
ext_fig_tp53_lollipop <- g1_tp %>%
    filter(Cohort != "Untreated", Gene == "TP53") %>%
    mutate(
        ID = DeID,
        PROTEIN_CHANGE = gene_aachange,
        PROTEIN_POS = as.numeric(sub(".*?([0-9]+).*", "\\1", sub(".*_", "", gene_aachange))),
        hit = case_when(
            Trilaciclib == "Placebo" ~ 1,
            Trilaciclib == "Trilaciclib" ~ 2
        ),
        gene = "TP53",
        mutation_type = VariantClass
    ) %>%
    my_lollipop(dat.genetics.unique=., 
                current.gene="TP53", 
                protein_domain=prot.info$protein_domain, 
                protein_length=prot.info$protein_length, 
                cutoff.hotspot = 20,
                colors.mutation_type=colors.mutation_type)
ggsave(paste0(figures_save_path, "ExtendedFigure1A.png"), plot = ext_fig_tp53_lollipop, width = 13, height = 8, dpi = 300)

prot.info = get_protein_info("PPM1D")
ext_fig_ppm1d_lollipop <- g1_tp %>%
    filter(Cohort != "Untreated", Gene == "PPM1D") %>%
    mutate(
        ID = DeID,
        PROTEIN_CHANGE = gene_aachange,
        PROTEIN_POS = as.numeric(sub(".*?([0-9]+).*", "\\1", sub(".*_", "", gene_aachange))),
        hit = case_when(
            Trilaciclib == "Placebo" ~ 1,
            Trilaciclib == "Trilaciclib" ~ 2
        ),
        gene = "PPM1D",
        mutation_type = VariantClass
    ) %>%
    my_lollipop(dat.genetics.unique=., 
                current.gene="PPM1D", 
                protein_domain=prot.info$protein_domain, 
                protein_length=prot.info$protein_length, 
                cutoff.hotspot = 20,
                colors.mutation_type=colors.mutation_type)
ggsave(paste0(figures_save_path, "ExtendedFigure1B.png"), plot = ext_fig_ppm1d_lollipop, width = 13, height = 8, dpi = 300)

prot.info = get_protein_info("CHEK2")
ext_fig_chek2_lollipop <- g1_tp %>%
    filter(Cohort != "Untreated", Gene == "CHEK2") %>%
    mutate(
        ID = DeID,
        PROTEIN_CHANGE = gene_aachange,
        PROTEIN_POS = as.numeric(sub(".*?([0-9]+).*", "\\1", sub(".*_", "", gene_aachange))),
        hit = case_when(
            Trilaciclib == "Placebo" ~ 1,
            Trilaciclib == "Trilaciclib" ~ 2
        ),
        gene = "CHEK2",
        mutation_type = VariantClass
    ) %>%
    my_lollipop(dat.genetics.unique=., 
                current.gene="CHEK2", 
                protein_domain=prot.info$protein_domain, 
                protein_length=prot.info$protein_length, 
                cutoff.hotspot = 20,
                colors.mutation_type=colors.mutation_type)
ggsave(paste0(figures_save_path, "ExtendedFigure1C.png"), plot = ext_fig_chek2_lollipop, width = 13, height = 8, dpi = 300)
```

# Extended Figure 1 Bubble Plot
```{r bubble}
df_to_plot_ddr <- g1_tp %>% 
    filter(
        Gene %in% DDR_genes,
        Cohort != "Untreated"
    ) %>%
    mutate(
        VariantClass = case_when(
            VariantClass == "missense_variant" ~ "Missense",
            VariantClass == "frameshift_variant" ~ "Frameshift",
            VariantClass == "inframe_insertion" ~ "Missense",
            VariantClass == "inframe_deletion" ~ "Missense",
            VariantClass == "splice_region_variant" ~ "Splicing",
            VariantClass == "start_lost" ~ "Nonsense",
            VariantClass == "stop_gained" ~ "Nonsense",
            VariantClass == "synonymous_variant" ~ "Synonymous",
            TRUE ~ "Other"
        ),
        VariantClass = factor(VariantClass, levels = c("Missense", "Frameshift", "Splicing", "Nonsense", "Synonymous", "Other"))
    )

ext_fig_tp53_bubble <- df_to_plot_ddr %>%
    filter(growth_rate != "Inf" & growth_rate != "-Inf") %>%
    filter(Gene == "TP53") %>%
    mutate(
        size_bin = case_when(
            heme_cosmic_count < 10 ~ "0-10",
            heme_cosmic_count >= 10 & heme_cosmic_count < 25 ~ "10-25",
            heme_cosmic_count >= 25 & heme_cosmic_count < 50 ~ "25-50",
            heme_cosmic_count >= 50 ~ ">50"
        ),
        size_bin = factor(size_bin, levels = c("0-10", "10-25", "25-50", ">50"))
    ) %>%
    ggplot(
        aes(
            x = preVAF,
            y = growth_rate,
            color = Cohort,
            size = size_bin,
            shape = VariantClass
        )
    ) +
    geom_point(alpha = 0.7) +
    scale_size_manual(values = c("0-10"= 6, "10-25"=8, "25-50"= 10, ">50"= 12)) +
    scale_shape_manual(values = c("Missense" = 16, "Frameshift" = 17, "Splicing" = 18, "Nonsense" = 15, "Synonymous" = 1, "Other" = 2)) +
    panel_theme_basic +
    labs(
        x = "VAF Pre-Treatment",
        y = "Growth Rate",
        title = "TP53",
        color = "Clinical Trial:",
        size = "Heme Cosmic Count:",
        shape = "Mutation Type:"
    ) +
    scale_fill_nejm() +
    scale_color_nejm() +
    scale_x_continuous(trans = scales::pseudo_log_trans(0.001), limit = c(NA, NA),  breaks = c(0, 0.01, 0.05, 0.01, 0.05, 0.1)) +
    scale_y_continuous(trans = scales::pseudo_log_trans(0.001), limit = c(NA, 0.05)) +
    theme(
        legend.position = c(0.60, 0.95),
        legend.box = "verticle",
        legend.title = element_text(size = 18),
        legend.text = element_text(size = 18),
        legend.spacing = unit(0.01, "cm"),
        plot.margin = margin(1, 1, 1, 1, "cm"),
    ) +
    guides(
        color = guide_legend(byrow = TRUE, order = 1, override.aes = list(size = 5)),
        shape = guide_legend(byrow = TRUE, order = 2, override.aes = list(size = 5)),
        size = guide_legend(byrow = TRUE, order = 3, override.aes = list(size = 5))
    )
ggsave(paste0(figures_save_path, "ExtendedFigure1D.png"), plot = ext_fig_tp53_bubble, width = 10, height = 7, dpi = 300)

ext_fig_ppm1d_bubble <- df_to_plot_ddr %>%
    filter(growth_rate != "Inf" & growth_rate != "-Inf") %>%
    filter(Gene == "PPM1D") %>%
    mutate(
        size_bin = case_when(
            heme_cosmic_count < 10 ~ "0-10",
            heme_cosmic_count >= 10 & heme_cosmic_count < 25 ~ "10-25",
            heme_cosmic_count >= 25 & heme_cosmic_count < 50 ~ "25-50",
            heme_cosmic_count >= 50 ~ ">50"
        ),
        size_bin = factor(size_bin, levels = c("0-10", "10-25", "25-50", ">50"))
    ) %>%
    ggplot(
        aes(
            x = preVAF,
            y = growth_rate,
            color = Cohort,
            size = size_bin,
            shape = VariantClass
        )
    ) +
    geom_point(alpha = 0.7) +
    scale_size_manual(values = c("0-10"= 6, "10-25"=8, "25-50"= 10, ">50"= 12)) +
    scale_shape_manual(values = c("Missense" = 16, "Frameshift" = 17, "Splicing" = 18, "Nonsense" = 15, "Synonymous" = 1, "Other" = 2)) +
    panel_theme_basic +
    labs(
        x = "VAF Pre-Treatment",
        y = "Growth Rate",
        title = "PPM1D",
        color = "Clinical Trial:",
        size = "Heme Cosmic Count:",
        shape = "Mutation Type:"
    ) +
    scale_fill_nejm() +
    scale_color_nejm() +
    scale_x_continuous(trans = scales::pseudo_log_trans(0.001), limit = c(NA, NA),  breaks = c(0, 0.01, 0.05, 0.01, 0.05, 0.1)) +
    scale_y_continuous(trans = scales::pseudo_log_trans(0.001), limit = c(NA, 0.05)) +
    theme(
        legend.position = "none",
        plot.margin = margin(1, 1, 1, 1, "cm"),
    )
ggsave(paste0(figures_save_path, "ExtendedFigure1E.png"), plot = ext_fig_ppm1d_bubble, width = 10, height = 7, dpi = 300)

ext_fig_chek2_bubble <- df_to_plot_ddr %>%
    filter(growth_rate != "Inf" & growth_rate != "-Inf") %>%
    filter(Gene == "CHEK2") %>%
    mutate(
        size_bin = case_when(
            heme_cosmic_count < 10 ~ "0-10",
            heme_cosmic_count >= 10 & heme_cosmic_count < 25 ~ "10-25",
            heme_cosmic_count >= 25 & heme_cosmic_count < 50 ~ "25-50",
            heme_cosmic_count >= 50 ~ ">50"
        ),
        size_bin = factor(size_bin, levels = c("0-10", "10-25", "25-50", ">50"))
    ) %>%
    ggplot(
        aes(
            x = preVAF,
            y = growth_rate,
            color = Cohort,
            size = size_bin,
            shape = VariantClass
        )
    ) +
    geom_point(alpha = 0.7) +
    scale_size_manual(values = c("0-10"= 6, "10-25"=8, "25-50"= 10, ">50"= 12)) +
    scale_shape_manual(values = c("Missense" = 16, "Frameshift" = 17, "Splicing" = 18, "Nonsense" = 15, "Synonymous" = 1, "Other" = 2)) +
    panel_theme_basic +
    labs(
        x = "VAF Pre-Treatment",
        y = "Growth Rate",
        title = "CHEK2",
        color = "Clinical Trial:",
        size = "Heme Cosmic Count:",
        shape = "Mutation Type:"
    ) +
    scale_fill_nejm() +
    scale_color_nejm() +
    scale_x_continuous(trans = scales::pseudo_log_trans(0.001), limit = c(NA, NA),  breaks = c(0, 0.01, 0.05, 0.01, 0.05, 0.1)) +
    scale_y_continuous(trans = scales::pseudo_log_trans(0.001), limit = c(NA, 0.05)) +
    theme(
        legend.position = "none",
        plot.margin = margin(1, 1, 1, 1, "cm"),
    )
ggsave(paste0(figures_save_path, "ExtendedFigure1F.png"), plot = ext_fig_chek2_bubble, width = 10, height = 7, dpi = 300)
ggsave(paste0(figures_save_path, "ExtendedFigure1G.png"), plot = ext_fig_tp53, width = 10, height = 7, dpi = 300)
```

# Supplementary Table 1 - CH Mutation Variants
```{r supp_table_1}
g1_df %>%
    filter(key != "" & key != "NA:NA:NA:NA") %>%
    mutate(Gene = factor(Gene, levels = gene_list)) %>%
    select(key, Gene) %>%
    distinct() %>%
    arrange(Gene) %>% 
    write.csv(paste0(figures_save_path, "SupplementaryTable1_CH_Mutations.csv"), row.names = FALSE)
```

# Supplementary Table 2 - Table Demographics
```{r supp_table_2}
table1(~ Age + Sex + Race + Ethnicity + SmokingStatus | Cohort + Trilaciclib, data = for_table %>% filter(whichdraw == "PreTx"))
```

# Supplementary Table 3 - Timepoints Variants
```{r supp_table_3}
g1_tp %>%
    select(DeID, Cohort, Trilaciclib, key, Gene, preVAF, duringVAF, postVAF, change_in_days) %>%
    mutate(
        Timepoint_1_VAF = preVAF,
        Timepoint_2_VAF = ifelse(Cohort == "TNBC", duringVAF, postVAF)
    ) %>%
    select(-preVAF, -duringVAF, -postVAF) %>%
    select(DeID, Cohort, Trilaciclib, key, Gene, Timepoint_1_VAF, Timepoint_2_VAF, change_in_days) %>%
    write.csv(paste0(figures_save_path, "SupplementaryTable3_Timepoints_Variants.csv"), row.names = FALSE)
```

# Supplementary Table 4 - Growth Rate Models with Cosmic Count
```{r supp_table_4}
g1_tp_tmp <- g1_tp %>% 
    filter(Trilaciclib != "Untreated") %>%
    filter(growth_rate != "Inf" & growth_rate != "-Inf")

model_growth_rate_cosmic_count_tp53 <- glm(
        formula = growth_rate ~ Trilaciclib + preVAF + CosmicCount + heme_cosmic_count + Cohort + Age + Sex + Race,
        data = g1_tp_tmp %>% filter(Gene == "TP53")
    ) %>% sjPlot::get_model_data(type = "est") %>%
    mutate(
        term = case_when(
            term == "TrilaciclibPlacebo" ~ "TrilaciclibVSPlacebo",
            term == "CohortTNBC" ~ "SCLCvsTNBC",
            term == "CohortCRC" ~ "SCLCvsCRC",
            TRUE ~ term
        )
    )
model_growth_rate_cosmic_count_tp53

model_growth_rate_cosmic_count_ppm1d <- glm(
        formula = growth_rate ~ Trilaciclib + preVAF + CosmicCount + heme_cosmic_count + Cohort + Age + Sex + Race,
        data = g1_tp_tmp %>% filter(Gene == "PPM1D")
    ) %>% sjPlot::get_model_data(type = "est") %>%
    mutate(
        term = case_when(
            term == "TrilaciclibPlacebo" ~ "TrilaciclibVSPlacebo",
            term == "CohortTNBC" ~ "SCLCvsTNBC",
            term == "CohortCRC" ~ "SCLCvsCRC",
            TRUE ~ term
        )
    )
model_growth_rate_cosmic_count_ppm1d

model_growth_rate_cosmic_count_chek2 <- glm(
        formula = growth_rate ~ Trilaciclib + preVAF + CosmicCount + heme_cosmic_count + Cohort + Age + Sex + Race,
        data = g1_tp_tmp %>% filter(Gene == "CHEK2")
    ) %>% sjPlot::get_model_data(type = "est") %>%
    mutate(
        term = case_when(
            term == "TrilaciclibPlacebo" ~ "TrilaciclibVSPlacebo",
            term == "CohortTNBC" ~ "SCLCvsTNBC",
            term == "CohortCRC" ~ "SCLCvsCRC",
            TRUE ~ term
        )
    )
model_growth_rate_cosmic_count_chek2

model_growth_rate_cosmic_count_ddr <- glm(
    formula = growth_rate ~ Trilaciclib + preVAF + CosmicCount + heme_cosmic_count + Cohort + Age + Sex + Race,
    data = g1_tp_tmp %>% filter(Gene_Class == "DDR")
) %>%
    sjPlot::get_model_data(type = "est") %>%
    mutate(
        term = case_when(
            term == "TrilaciclibPlacebo" ~ "TrilaciclibVSPlacebo",
            term == "CohortTNBC" ~ "SCLCvsTNBC",
            term == "CohortCRC" ~ "SCLCvsCRC",
            TRUE ~ term
        )
    )
model_growth_rate_cosmic_count_ddr
```

# Supplemental Table 5 - Blood Counts Analysis
```{r blood_counts}
g1_tp_tmp <- g1_tp %>%
    filter(Cohort != "Untreated") %>%
    filter(growth_rate != "Inf" & growth_rate != "-Inf") %>%
    mutate(Cohort = factor(Cohort, levels = c("SCLC", "TNBC", "CRC"))) %>%
    mutate(
        VAF_1 = preVAF * 100,
        VAF_2 = case_when(
            Cohort == "TNBC" ~ duringVAF * 100,
            TRUE ~ postVAF * 100
        ),
        log_VAF_1 = log(VAF_1),
        log_VAF_2 = log(VAF_2),
    ) %>%
    filter(VAF_1 > 0 & VAF_2 > 0)

model_change_in_prop_blood_counts_gene_class <- map_dfr(c("DTA", "DDR"), function(gene_class){
    print(gene_class)
    glm(
        formula = VAF_2/VAF_1 ~ change_in_prop_MONO + change_in_prop_LYM + change_in_prop_GRAN + Cohort + Trilaciclib,
        data = g1_tp_tmp %>% filter(Gene_Class == gene_class)
    ) %>%
    sjPlot::get_model_data(type = "est") %>%
    mutate(Gene_Class = gene_class)
})
model_change_in_prop_blood_counts_gene_class
```

# Supplemental Figure 1 - Percent Change in VAF per Months
```{r supp_fig1, fig.width=10, fig.height=10}
data_to_plot <- g1_tp %>%
    filter(growth_rate != "Inf" & growth_rate != "-Inf") %>%
    filter(Gene_Class == "DDR")

data_to_plot %>%
    filter(Gene_Class == "DDR") %>%
    group_by(Trilaciclib) %>%
    summarise(
        median_gr = median(growth_rate),
        average_gr = mean(growth_rate),
    ) %>%
    pivot_wider(names_from = Trilaciclib, values_from = c(median_gr, average_gr)) %>%
    mutate(perc_reduction = (1 - median_gr_Trilaciclib / median_gr_Placebo) * 100) %>%
    select(perc_reduction)

data_to_plot <- data_to_plot %>%
    group_by(DeID, Gene) %>%
    slice_max(abs(change_in_VAF), n = 1) %>%
    ungroup() %>%
    mutate(
        starting_clone_vaf = case_when(
            preVAF < 0.005 ~ "<0.5%",
            preVAF >= 0.005 & preVAF < 0.05 ~ "0.5-2%",
            preVAF >= 0.02 ~ ">=2%"
        ),
        starting_clone_vaf = factor(starting_clone_vaf, levels = c("<0.5%", "0.5-2%", ">=2%")),
        treatment_time = case_when(
            change_in_days < 30 ~ "1 Month",
            change_in_days >= 30 & change_in_days < 90 ~ "3 Months",
            change_in_days >= 90 & change_in_days < 180 ~ "Half a Year",
            change_in_days >= 180 & change_in_days < 360 ~ "A Year",
            change_in_days >= 360 ~ "Over A Year"
        ),
        treatment_time = factor(treatment_time, levels = c("1 Month", "3 Months", "Half a Year", "A Year")),
        alpha_val = case_when(
            abs(growth_rate) < 0.01 ~ 0.3,
            abs(growth_rate) >= 0.01 & abs(growth_rate) < 0.05 ~ 0.5,
            abs(growth_rate) >= 0.05 ~ 1
        )
    ) %>%
    mutate(
        change_in_months = change_in_days / 30,
        change_in_VAF_per_month = change_in_VAF / change_in_months,
        change_in_VAF_per_day = change_in_VAF / change_in_days,
    )

supp_fig1 <- data_to_plot %>%
    ggplot(
        aes(
            x = Trilaciclib,
            y = change_in_VAF_per_month,
            color = Trilaciclib,
            size = starting_clone_vaf,
            alpha = alpha_val
        )
    ) +
    geom_hline(yintercept = 0, linetype = "dashed", alpha = 0.5) +
    geom_boxplot(position = position_dodge(width = 0.8), alpha = 0.1, size = 1, outlier.shape = NA) +
    geom_point(position = position_jitterdodge(jitter.width=1)) +
    #facet_wrap(~Gene_Class) +
    #geom_point(position = position_dodge(width = 0.8), alpha = 0.3) +
    labs(
        x = "",
        y = "Change in VAF per Month",
        color = "Trilaciclib",
        size = "Starting Clone VAF",
    ) +
    panel_theme_basic +
    #scale_fill_nejm() +
    #scale_color_nejm() +
    scale_fill_manual(values = treatment_colors) +
    scale_color_manual(values = treatment_colors) +
    scale_alpha_identity() +
    scale_y_continuous(
        trans = scales::pseudo_log_trans(0.0001),
        limits = \(x) {c(-max(abs(x)), max(abs(x)))},
        #breaks = scales::extended_breaks(n = 10),
        breaks = c(-0.1, -0.05, -0.01, -0.005, -0.001, 0, 0.001, 0.005, 0.01, 0.05, 0.1),
        labels = scales::percent_format(accuracy = 0.1)
    ) +
    theme(
        legend.position = "right",
        legend.direction = "vertical",
        legend.title = element_text(size = 18),
        legend.text = element_text(size = 18)
    ) +
    guides(
        color = guide_legend(title.position = "top", title = "Drug Treatment"),
        size = guide_legend(title.position = "top")
    )
supp_fig1
ggsave(paste0(figures_save_path, "SupplementalFigure1.png"), plot = supp_fig1, width = 10, height = 8, dpi = 300)

data_to_plot %>%
    ungroup() %>%
    group_by(Trilaciclib) %>%
    summarise(
        mean_change_in_VAF = median(change_in_VAF),
        mean_change_in_VAF_per_month = median(change_in_VAF_per_month),
        change_in_days = median(change_in_days)
    )
```

# Supplemental Figure 2 - VAF >= 0.2%
```{r supp_fig_2, fig.width=10, fig.height=10}
# SCLC Trial
df_to_plot <- g1_tp %>%
    filter(Cohort == "SCLC" | Cohort == "Untreated") %>%    # Filter for SCLC and Untreated cohorts
    filter(Gene_Class == "DTA" | Gene_Class == "DDR") %>%   # Filter for DTA and DDR gene classes
    filter(growth_rate != "Inf" & growth_rate != "-Inf")    # Remove Inf and -Inf growth rates

df_to_plot <- df_to_plot %>%
    filter(!(preVAF < 0.002 & postVAF < 0.002))

model_growth_rate_class <- model_growth_rate_class_fun(df_to_plot)

# Absolute DDR Growth Rate Calculations
df_to_plot %>%
    filter(Gene_Class == "DDR") %>%
    group_by(Trilaciclib) %>%
    summarise(
        median_gr = median(growth_rate),
        average_gr = mean(growth_rate),
    ) %>%
    pivot_wider(names_from = Trilaciclib, values_from = c(median_gr, average_gr)) %>%
    mutate(perc_reduction = (1 - median_gr_Trilaciclib / median_gr_Placebo) * 100) %>%
    select(perc_reduction)

counts <- df_to_plot %>%
    group_by(Trilaciclib, Gene_Class) %>%
    summarise(n = n(), .groups = "drop")

supp_fig2a <- df_to_plot %>%
    ggplot(
        aes(
            x = Gene_Class,                 # Notice how we are using the Gene_Class variable for the x-axis where before we were using Trilaciclib.
            y = growth_rate, 
            fill = Trilaciclib              # We will still group data by Trilaciclib, but our main focus is on the Gene_Class variable.
        )
    ) +
    geom_hline(yintercept = 0, color = "black", alpha = 0.3) +
    geom_boxplot(outlier.shape = NA) +
    geom_point(position = position_jitterdodge(jitter.width=0.2), alpha = 0.3) +
    geom_text(data = counts, aes(label = n, y = 0.06), size = 8, position = position_dodge(width = 0.8)) +
    labs(
        y = "Growth Rate", 
        x = '', 
        title = ""
    ) +
    panel_theme_basic +
    theme(plot.margin = margin(-1, 0, -1, 0, "cm")) +
    scale_y_continuous(trans = scales::pseudo_log_trans(0.001), breaks = c(-0.05, -0.01, -0.005, 0, 0.005, 0.01, 0.05), limit = c(NA, 1.5)) +
    scale_fill_manual(values = treatment_colors) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 5, 1.75, 2, model_growth_rate_class$p.value[3], model_growth_rate_class$p.stars[3], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 6, 1.75, 2.25, model_growth_rate_class$p.value[4], model_growth_rate_class$p.stars[4], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 7, 2, 2.25, model_growth_rate_class$p.value[6], model_growth_rate_class$p.stars[6], show_values = FALSE))
    # do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 5, 0.75, 1, model_growth_rate_class$p.value[1], model_growth_rate_class$p.stars[1], show_values = FALSE)) +
    # do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 6, 0.75, 1.25, model_growth_rate_class$p.value[2], model_growth_rate_class$p.stars[2], show_values = FALSE)) +
    # do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 7, 1, 1.25, model_growth_rate_class$p.value[5], model_growth_rate_class$p.stars[5], show_values = FALSE))
supp_fig2a
ggsave(paste0(figures_save_path, "SupplementalFigure2A.png"), plot = supp_fig2a, width = 10, height = 8, dpi = 300)

# CRC Trial
df_to_plot <- g1_tp %>% 
    filter(Cohort == "CRC" | Cohort == "Untreated") %>%     # Filter for CRC and Untreated cohorts
    filter(Gene_Class == "DTA" | Gene_Class == "DDR") %>%   # Filter for DTA and DDR gene classes
    filter(growth_rate != "Inf" & growth_rate != "-Inf")    # Remove Inf and -Inf growth rates

df_to_plot <- df_to_plot %>%
    filter(!(preVAF < 0.002 & postVAF < 0.002))

model_growth_rate_class <- model_growth_rate_class_fun(df_to_plot)

# Absolute DDR Growth Rate Calculations
df_to_plot %>%
    filter(Gene_Class == "DDR") %>%
    group_by(Trilaciclib) %>%
    summarise(
        median_gr = median(growth_rate),
        average_gr = mean(growth_rate),
    ) %>%
    pivot_wider(names_from = Trilaciclib, values_from = c(median_gr, average_gr)) %>%
    mutate(perc_reduction = (1 - median_gr_Trilaciclib / median_gr_Placebo) * 100) %>%
    select(perc_reduction)

counts <- df_to_plot %>%
    group_by(Trilaciclib, Gene_Class) %>%
    summarise(n = n(), .groups = "drop")

supp_fig2b <- df_to_plot %>%
    filter(growth_rate != "Inf" & growth_rate != "-Inf") %>%
    filter(Gene_Class == "DDR" | Gene_Class == "DTA") %>%       # We are filtering the data to only include the DDR and DTA gene classes.
    ggplot(
        aes(
            x = Gene_Class,                 # Notice how we are using the Gene_Class variable for the x-axis where before we were using Trilaciclib.
            y = growth_rate, 
            fill = Trilaciclib              # We will still group data by Trilaciclib, but our main focus is on the Gene_Class variable.
        )
    ) +
    geom_hline(yintercept = 0, color = "black", alpha = 0.3) +
    geom_boxplot(outlier.shape = NA) +
    geom_point(position = position_jitterdodge(jitter.width=0.2), alpha = 0.3) +
    geom_text(data = counts, aes(label = n, y = 0.06), size = 8, position = position_dodge(width = 0.8)) +
    labs(
        y = "Growth Rate", 
        x = '', 
        title = ""
    ) +
    panel_theme_basic +
    theme(plot.margin = margin(-1, 0, -1, 0, "cm")) +
    scale_y_continuous(trans = scales::pseudo_log_trans(0.001), breaks = c(-0.05, -0.01, -0.005, 0, 0.005, 0.01, 0.05), limit = c(NA, 1.5)) +
    scale_fill_manual(values = treatment_colors) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 5, 1.75, 2, model_growth_rate_class$p.value[3], model_growth_rate_class$p.stars[3], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 6, 1.75, 2.25, model_growth_rate_class$p.value[4], model_growth_rate_class$p.stars[4], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 7, 2, 2.25, model_growth_rate_class$p.value[6], model_growth_rate_class$p.stars[6], show_values = FALSE))
    # do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 5, 0.75, 1, model_growth_rate_class$p.value[1], model_growth_rate_class$p.stars[1], show_values = FALSE)) +
    # do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 6, 0.75, 1.25, model_growth_rate_class$p.value[2], model_growth_rate_class$p.stars[2], show_values = FALSE)) +
    # do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 7, 1, 1.25, model_growth_rate_class$p.value[5], model_growth_rate_class$p.stars[5], show_values = FALSE)) +
supp_fig2b
ggsave(paste0(figures_save_path, "SupplementalFigure2B.png"), plot = supp_fig2b, width = 10, height = 8, dpi = 300)

# TNBC Trial
df_to_plot <- g1_tp %>%
    filter(Cohort == "TNBC" | Cohort == "Untreated") %>%    # Filter for TNBC and Untreated cohorts
    filter(Gene_Class == "DTA" | Gene_Class == "DDR") %>%   # Filter for DTA and DDR gene classes
    filter(growth_rate != "Inf" & growth_rate != "-Inf")    # Remove Inf and -Inf growth rates

df_to_plot <- df_to_plot %>%
    filter(!(preVAF < 0.002 & postVAF < 0.002))

model_growth_rate_class <- model_growth_rate_class_fun(df_to_plot)

df_to_plot %>%
    filter(Gene_Class == "DDR") %>%
    group_by(Trilaciclib) %>%
    summarise(
        median_gr = median(growth_rate),
        average_gr = mean(growth_rate),
    ) %>%
    pivot_wider(names_from = Trilaciclib, values_from = c(median_gr, average_gr)) %>%
    mutate(perc_reduction = (1 - median_gr_Trilaciclib / median_gr_Placebo) * 100) %>%
    select(perc_reduction)

counts <- df_to_plot %>%
    group_by(Trilaciclib, Gene_Class) %>%
    summarise(n = n(), .groups = "drop")

supp_fig2c <- df_to_plot %>%
    ggplot(
        aes(
            x = Gene_Class,                 # Notice how we are using the Gene_Class variable for the x-axis where before we were using Trilaciclib.
            y = growth_rate, 
            fill = Trilaciclib              # We will still group data by Trilaciclib, but our main focus is on the Gene_Class variable.
        )
    ) +
    geom_hline(yintercept = 0, color = "black", alpha = 0.3) +
    geom_boxplot(outlier.shape = NA) +
    geom_point(position = position_jitterdodge(jitter.width=0.2), alpha = 0.3) +
    geom_text(data = counts, aes(label = n, y = 0.06), size = 8, position = position_dodge(width = 0.8)) +
    labs(
        y = "Growth Rate", 
        x = '', 
        title = ""
    ) +
    panel_theme_basic +
    theme(plot.margin = margin(-1, 0, -1, 0, "cm")) +
    scale_y_continuous(trans = scales::pseudo_log_trans(0.001), breaks = c(-0.05, -0.01, -0.005, 0, 0.005, 0.01, 0.05), limit = c(NA, 1.5)) +
    scale_fill_manual(values = treatment_colors) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 5, 1.75, 2, model_growth_rate_class$p.value[3], model_growth_rate_class$p.stars[3], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 6, 1.75, 2.25, model_growth_rate_class$p.value[4], model_growth_rate_class$p.stars[4], show_values = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DDR"), 7, 2, 2.25, model_growth_rate_class$p.value[6], model_growth_rate_class$p.stars[6], show_values = FALSE))
    # do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 5, 0.75, 1, model_growth_rate_class$p.value[1], model_growth_rate_class$p.stars[1], show_values = FALSE)) +
    # do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 6, 0.75, 1.25, model_growth_rate_class$p.value[2], model_growth_rate_class$p.stars[2], show_values = FALSE)) +
    # do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene_Class == "DTA"), 7, 1, 1.25, model_growth_rate_class$p.value[5], model_growth_rate_class$p.stars[5], show_values = FALSE))
supp_fig2c
ggsave(paste0(figures_save_path, "SupplementalFigure2C.png"), plot = supp_fig2c, width = 10, height = 8, dpi = 300)
```

# Supplemental Figure 2 - Genes Specific
```{r supp_fig_2_genes, fig.width=10, fig.height=7}
df_to_plot <- g1_tp %>% 
    filter(Gene_Class == "DDR") %>%                             # Filter for DDR gene class
    filter(growth_rate != "Inf" & growth_rate != "-Inf") %>%    # Remove Inf and -Inf growth rates
    mutate(Variant_Classification = if_else(VariantClass == "missense_variant", "Missense", "Nonsense"))

df_to_plot <- df_to_plot %>%
    filter(!(preVAF < 0.002 & postVAF < 0.002))

model_growth_rate_gene <- purrr::cross_df(list(
    gene = c("TP53", "PPM1D", "CHEK2"),
    comparison = c("all", "no_untreated")
)) %>%
    pmap_dfr(function(gene, comparison) {
        print(paste(gene, comparison))
        if (comparison == "all") {
            df_subset <- df_to_plot %>% filter(Gene == gene)
            model <- glm(
                formula = growth_rate ~ Trilaciclib + Age,
                data = df_subset
            ) %>%
                sjPlot::get_model_data(type = "est") %>%
                filter(!grepl("Age", term), !grepl("Cohort", term)) %>%
                mutate(term = ifelse(term == "TrilaciclibTrilaciclib", "UntreatedVSTrilaciclib", "UntreatedVSPlacebo"))
        } else if (comparison == "no_untreated") {
            df_subset <- df_to_plot %>% filter(Gene == gene & Trilaciclib != "Untreated")
            model <- glm(
                formula = growth_rate ~ Trilaciclib + Cohort + Age,
                data = df_subset
            ) %>%
                sjPlot::get_model_data(type = "est") %>%
                filter(!grepl("Age", term), !grepl("Cohort", term)) %>%
                mutate(term = "TrilaciclibVSPlacebo")
        }
        model %>% mutate(gene = gene, comparison = comparison)
    })

glm(
    formula = growth_rate ~ Trilaciclib + Age + Cohort,
    data = df_to_plot %>% filter(Gene == "TP53", Trilaciclib != "Untreated")
) %>%
    sjPlot::get_model_data(type = "est")

counts_genes <- df_to_plot %>%
    group_by(Trilaciclib, Gene) %>%
    summarise(n = n(), .groups = "drop")

supp_fig2d <- df_to_plot %>%
    ggplot(
        aes(
            x = Gene,
            y = growth_rate, 
            fill = Trilaciclib
        )
    ) +
    geom_hline(yintercept = 0, color = "black", alpha = 0.3) +
    geom_boxplot(outlier.shape = NA) +
    geom_point(position = position_jitterdodge(jitter.width=0.2), alpha = 0.3) +
    geom_text(data = counts_genes, aes(label = n, y = 0.06), size = 6, position = position_dodge(width = 0.8)) +
    labs(
        y = "Growth Rate", 
        x = '', 
        title = ""
    ) +
    panel_theme_basic +
    theme(
        plot.margin = margin(-1, 0, -1, 0, "cm")
    ) +
    scale_y_continuous(trans = scales::pseudo_log_trans(0.001), breaks = c(-0.05, -0.01, -0.005, 0, 0.005, 0.01, 0.05), limit = c(NA, 1.5)) +
    scale_fill_manual(values = treatment_colors) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "TP53"), 5, 0.75, 1, model_growth_rate_gene$p.value[1], model_growth_rate_gene$p.stars[1], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "TP53"), 6, 0.75, 1.25, model_growth_rate_gene$p.value[2], model_growth_rate_gene$p.stars[2], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "TP53"), 7, 1, 1.25, model_growth_rate_gene$p.value[7], model_growth_rate_gene$p.stars[7], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "PPM1D"), 5, 1.75, 2, model_growth_rate_gene$p.value[3], model_growth_rate_gene$p.stars[3], show_values = FALSE, hide_ns = FALSE)) + 
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "PPM1D"), 6, 1.75, 2.25, model_growth_rate_gene$p.value[4], model_growth_rate_gene$p.stars[4], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "PPM1D"), 7.1, 2, 2.25, model_growth_rate_gene$p.value[8], model_growth_rate_gene$p.stars[8], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "CHEK2"), 5, 2.75, 3, model_growth_rate_gene$p.value[5], model_growth_rate_gene$p.stars[5], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "CHEK2"), 6, 2.75, 3.25, model_growth_rate_gene$p.value[6], model_growth_rate_gene$p.stars[6], show_values = FALSE, hide_ns = FALSE)) +
    do.call(geom_signif, default_sig_data(df_to_plot %>% filter(Gene == "CHEK2"), 7.1, 3, 3.25, model_growth_rate_gene$p.value[9], model_growth_rate_gene$p.stars[9], show_values = FALSE, hide_ns = FALSE)) 
supp_fig2d
ggsave(paste0(figures_save_path, "SupplementalFigure2D.png"), plot = supp_fig2d, width = 10, height = 8, dpi = 300)
```

# Supplemental Figure 3 - Bioinformatics Figure 4 Validation
```{r supp_fig_3, fig.width=10, fig.height=7}
# Calculate Pearson correlation for all points
pearson_corr <- cor(bio_fig4_data$rep1, bio_fig4_data$rep2, use = "complete.obs", method = "pearson")

axis_ticks <- c(0.0001, 0.0002, 0.0005, 0.001, 0.0020, 0.005, 0.01, 0.020, 0.05, 0.1)

supp_fig3 <- bio_fig4_data %>%
    mutate(
        depth_bin = case_when(
            average_depth_1 < 15000 | average_depth_2 < 15000 ~ "<15k",
            (average_depth_1 >= 15000 & average_depth_1 < 16000) | (average_depth_2 >= 15000 & average_depth_2 < 16000) ~ "15k-16k",
            (average_depth_1 >= 16000 & average_depth_1 < 17000) | (average_depth_2 >= 16000 & average_depth_2 < 17000) ~ "16k-17k",
            (average_depth_1 >= 17000 & average_depth_1 < 18000) | (average_depth_2 >= 17000 & average_depth_2 < 18000) ~ "17k-18k",
            (average_depth_1 >= 18000 & average_depth_1 < 19000) | (average_depth_2 >= 18000 & average_depth_2 < 19000) ~ "18k-19k",
            (average_depth_1 >= 19000 & average_depth_1 < 20000) | (average_depth_2 >= 19000 & average_depth_2 < 20000) ~ "19k-20k",
            average_depth_1 >= 20000 | average_depth_2 >= 20000 ~ ">=20k"
        )
    ) %>%
    ggplot(
        aes(
            x = rep1,
            y = rep2,
            #color = depth_bin,
            #shape = PassOrNot
        )
    ) +
    geom_point(size = 4) +
    coord_fixed(clip = "off") +
    geom_abline(linetype = "dashed") +
    labs(
        title = "",
        color = "Reason", 
        x = "Replicate 1 - VAF",
        y = "Replicate 2 - VAF"
    ) +
    panel_theme_basic +
    theme(
        panel.grid.major = element_line(),
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size = 18, angle = 45, hjust = 1),
        axis.text.y = element_text(size = 18),
        plot.margin = margin(0, 0, 0, 0, "cm")
    ) +
    #scale_shape_manual(values=c(1, 15, 16, 17, 18)) +
    scale_color_nejm() +
    scale_x_log10(
        limits = c(0.0001, 0.1),
        breaks = axis_ticks, 
        labels = scales::percent_format(accuracy = 0.01)
    ) +
    scale_y_log10(
        limits = c(0.0001, 0.1),
        breaks = axis_ticks, 
        labels = scales::percent_format(accuracy = 0.01)
    ) +
    annotate("text", x = 0.0001, y = 0.055, label = paste0("Pearson r = ", round(pearson_corr, 3)), size = 6, hjust = 0)
supp_fig3
ggsave(paste0(figures_save_path, "SupplementalFigure3.png"), plot = supp_fig3, width = 10, height = 8, dpi = 300)
```